propagation/w3c.go

```diff
@@ -32,6 +38,14 @@ func MarshalW3CTraceContext(ctx context.Context, prop *PropagationContext) (cont
 // UnmarshalW3CTraceContext parses the information provided in the appropriate headers
 // and creates a PropagationContext instance. Headers are passed in via an HTTPSupplier,
 // which is an interface that defines Get and Set methods, http.Header is an implementation.
+//
+// Context is passed into this function and returned so that we can maintain the value of the
+// tracestate header. This is required in order to use the Propagator interface exported by the
+// OpenTelemetry Go SDK and avoid writing our own W3C Trace Context parser and serializer.
+//
+// If the headers are missing or empty strings, the propagation object will have zero values.
+// In the case of Span IDs and Trace IDs, they will be populated with 16 and 32 character strings
+// containing all zeroes.
 func UnmarshalW3CTraceContext(ctx context.Context, headers map[string]string) (context.Context, *PropagationContext) {
```

Comment:
the otel `trace` package does have an [`IsValid()`](https://pkg.go.dev/go.opentelemetry.io/otel/api/trace?tab=doc#SpanContext.IsValid) function that hangs off a `spanContext`. Do you want this unmarshal function to match the others in "if the trace context is invalid (eg it doesn't have a trace ID but does have other stuff) it will return an error"? Equally reasonable to put that responsibility on the caller.